src/ui/components/App.tsx

```diff
@@ -1,60 +1,93 @@
-import { useRef, useCallback } from 'preact/hooks';
-
+import { useRef, useCallback, useState, useEffect } from 'preact/hooks';
 import { Uri } from 'monaco-editor';
 
 import Editor from './Editor';
+import JS from './JS';
+import Menu from './Menu';
 import Preview from './Preview';
 import StatusBar from './StatusBar';
 import useMonaco from '../hooks/useMonaco';
 import useEsbuildWorker from '../hooks/useEsbuildWorker';
 import useAstroWorker from '../hooks/useAstroWorker';
+import { TABS } from '../const';
+import useWindowSize from '../hooks/useWindowSize';
 
 export interface Props {
-  Monaco: typeof import('../../editor/modules/monaco')
-  esbuildWorker: Worker
-  astroWorker: Worker
+  Monaco: typeof import('../../editor/modules/monaco');
+  esbuildWorker: Worker;
+  astroWorker: Worker;
 }
 
-const LETTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split('');
+let resizeTimeout: number | undefined;
+
+const LETTERS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
 const App: FunctionalComponent<Props> = ({ Monaco, esbuildWorker, astroWorker }) => {
-  const editorRef = useRef<HTMLElement|null>(null);
+  const editorRef = useRef<HTMLElement | null>(null);
   const { editor, models, currentModel, value, setTab, addTab, removeTab } = useMonaco(Monaco, editorRef);
   const { html, error } = useEsbuildWorker(esbuildWorker, editor, [value]);
   const { js, duration } = useAstroWorker(astroWorker, editor, [value]);
+  const { isDesktop } = useWindowSize();
+  const [currentTab, setCurrentTab] = useState<TabName>(isDesktop ? TABS.PREVIEW : TABS.CODE);
 
   const onAddTab = useCallback(() => {
-    const basenames = models.map(model => model.uri.path.split('/').pop().slice(0, '.astro'.length * -1)).filter(c => c.startsWith('Component'));
+    const basenames = models
+      .map((model) =>
+        model.uri.path
+          .split('/')
+          .pop()
+          .slice(0, '.astro'.length * -1)
+      )
+      .filter((c) => c.startsWith('Component'));
     let filename = 'Component';
-    console.log(basenames);
     for (const basename of basenames) {
       if (basename === filename) {
         if (filename === 'Component') {
           filename = 'Component1';
         } else {
           const number = Number.parseInt(filename.replace(/^Component/, '')) + 1;
-          filename = `Component${number}`
+          filename = `Component${number}`;
         }
       }
     }
-    addTab(`---
+    addTab(
+      `---
 const name = "Component"
 ---
 
-<h1>Hello {name}</h1>`, 'astro', Uri.parse(`inmemory://model/src/components/${filename}.astro`))
-  }, [models])
+<h1>Hello {name}</h1>`,
+      'astro',
+      Uri.parse(`inmemory://model/src/components/${filename}.astro`)
+    );
+  }, [models]);
 
   const onSetTab = (model) => {
     console.log(model.uri.toString());
     setTab(model);
-  }
+  };
+
+  // confirm exit before navigating
+  useEffect(() => {
+    window.onbeforeunload = () => confirm(`Exit Astro Play? Your work will be lost!`);
```

Comment:
Adds an exit confirmation. At least until we get localStorage saving in place (TODO; next PR)